packages/atomic/cypress/integration/facet-standard.cypress.ts

```diff
@@ -0,0 +1,406 @@
+import {getAnalyticsAt, getApiRequestBodyAt} from '../utils/network';
+import {setUpPage, shouldRenderErrorComponent} from '../utils/setupComponent';
+import {
+  FacetSelectors,
+  createAliasShadow,
+  createAliasFacetUL,
+  createBreadcrumbShadowAlias,
+} from './facet-selectors';
+
+const facetProp = {
+  field: 'author',
+  label: 'Authors',
+};
+
+const compareAlphanumericalValues = (valueA: string, valueB: string) => {
+  if (valueA.toLowerCase() < valueB.toLowerCase()) {
+    return -1;
+  }
+  if (valueA.toLowerCase() > valueB.toLowerCase()) {
+    return 1;
+  }
+  return 0;
+};
+
+const sortCriteriaOption = {
+  automatic: 'automatic',
+  alphanumeric: 'alphanumeric',
+  occurrences: 'occurrences',
+  score: 'score',
+};
+
+function setupPager(field: string, label: string, option?: string) {
+  setUpPage(
+    `<atomic-breadcrumb-manager></atomic-breadcrumb-manager>
+     <atomic-facet field="${field}" label="${label}" ${option}></atomic-facet>`
+  );
+}
+
+function componentLoaded(label: string) {
+  cy.get(FacetSelectors.facetStandard).should('be.visible');
+  cy.get('@facetShadow').find('div:nth-child(1)').should('contain.text', label);
+  cy.checkA11y(FacetSelectors.facetStandard);
+}
+
+function validateNumberofValue(totalNumber: number) {
+  cy.wait(200);
+  cy.get('@facet_ul').find('li').its('length').should('eq', totalNumber);
+}
+
+async function facetValueShouldDisplayInBreadcrumb(
+  facetValueSelector: string,
+  valueDisplayInBreadcrumbSelector: string
+) {
+  cy.get(facetValueSelector)
+    .find('label span:nth-child(1)')
+    .invoke('text')
+    .then((text) => {
+      cy.get('@breadcrumbFacet')
+        .find(valueDisplayInBreadcrumbSelector)
+        .should('be.visible')
+        .contains(text);
+    });
+}
+
+async function validateBasicFacetFunctionality(
+  selector: string,
+  field: string
+) {
+  cy.get(selector).click();
+  cy.get(selector).find(FacetSelectors.checkbox).should('be.checked');
+  const analyticsBody = (await getAnalyticsAt('@coveoAnalytics', 1)).request
+    .body;
+  expect(analyticsBody).to.have.property('actionCause', 'facetSelect');
+  expect(analyticsBody.customData).to.have.property('facetField', field);
+  expect(analyticsBody.facetState[0]).to.have.property('state', 'selected');
+  expect(analyticsBody.facetState[0]).to.have.property('field', field);
+}
+async function getTextOfAllElements(selector: string) {
+  return new Promise((resolve) => {
+    cy.get(selector).then((elems) => {
+      const originalValues = [...elems].map((el: any) => el.textContent.trim());
```

Comment:
Why do you need to use the spread operators on `elems` ? Is not not an array already ?